File: old_core/demos/chromatic.py
from core_py import LocalPlayer, WaveSynth, Note
import core_py

p = LocalPlayer()

wg = WaveSynth()

note_length = core_py.sample_rate//8
note_intensity = core_py.max_intensity//4
notes = [Note(i, -1, note_intensity, note_length*i, note_length) for i in range(50)]

for note in notes:
    wg.add_note(note)

p.add_instrument(wg)
p.play()

File: old_core/demos/chromatic.cpp
#include <cmath>
#include <iostream>
#include <stdexcept>
#include <memory>
#include "../LocalPlayer.h"
#include "../Instrument.h"
#include "../WaveSynth.h"

int main(int argc, char ** argv)
{
    PlayerHandle p = std::make_shared<LocalPlayer>();
    InstrumentHandle wg = std::make_shared<WaveSynth>("data");

    Beat note_length(1);
    Sample note_intensity = Sample::max_intensity;
    WaveSynth::Notes notes;
    for(int i = 0; i < 80; ++i) {
        notes.push_back(
            Note(i,-1,note_intensity,note_length*i,note_length));
    }

    wg->add_notes(notes);
    p->add_instrument(wg);
    p->play();
}
